### PR TITLE
DTFS2-4586 : ACBS Facility update

### PR DESCRIPTION
**Pull request by abhi-markan**
ACBS facility stage update issue has been patched, this commit entails following changes:

* Addition of additional `dealType` argument used for facility type and exposure period computation.
* ACBS MDM EA calls - Safe array index access condition.
* ACBS Facility update - Addition of `exposurePeriod` and `description` post fields for an accurate facility record.
* `issuedFacilityMaster`  sub object existence verification. 
* GEF Facility mapping - `ukefFacilityId` as string
* Introduction of try-catch blocks on various calls.
* Improved documentation.
* Lint fixes.

### CODE REVIEW

File: trade-finance-manager-api/src/v1/controllers/update-issued-facilities.js [hunk 1]
```diff
@@ -28,7 +28,10 @@ const updatedIssuedFacilities = async (deal) => {
       hasBeenAcknowledged,
     } = facility;
 
-    const shouldUpdateFacility = (hasBeenIssued && !hasBeenAcknowledged);
+    // `hasBeenAcknowledged` is BSS/EWCS specific only
+    const shouldUpdateFacility = dealType === CONSTANTS.DEALS.DEAL_TYPE.GEF
```
Comment (IgnatG) [inline trade-finance-manager-api/src/v1/controllers/update-issued-facilities.js:32]: I'm not sure if you need this here, but we have now `hasBeenIssued` both for GEF and BSS

File: trade-finance-manager-api/src/v1/controllers/acbs.controller.js [hunk 2]
```diff
@@ -62,41 +65,43 @@ const updateIssuedFacilityAcbs = ({ facilityId, issuedFacilityMaster }) =>
...
-
-        default:
-          await updateDealAcbs(task.output);
+    const collection = await db.getCollection('durable-functions-log');
+    const runningTasks = await collection.find({
+      type: 'ACBS',
+      status: 'Running',
+    }).toArray();
+    const tasks = await runningTasks.map(async ({ acbsTaskLinks = {} }) =>
+      api.getFunctionsAPI(CONSTANTS.DURABLE_FUNCTIONS.TYPE.ACBS, acbsTaskLinks.statusQueryGetUri));
```
Comment (ttbarnes) [inline trade-finance-manager-api/src/v1/controllers/acbs.controller.js:76]: Nice - was the issue here that `acbsTaskLinks.statusQueryGetUri` wasn't returning a uri or invalid one?

File: trade-finance-manager-api/src/v1/controllers/update-issued-facilities.js (hunk 1 above)
Comment (abhi-markan) [reply trade-finance-manager-api/src/v1/controllers/update-issued-facilities.js:32]: Brilliant, have updated.

File: trade-finance-manager-api/src/v1/controllers/acbs.controller.js (hunk 2 above)
Comment (abhi-markan) [reply trade-finance-manager-api/src/v1/controllers/acbs.controller.js:76]: The issue was first argument was missing and regex needs ternary operator for safety.

**Review by IgnatG — APPROVED** [review]
LGTM

**Review by IgnatG — APPROVED** [review]

**Pull request merged**